test/api.spec.js

```diff
@@ -0,0 +1,62 @@
+import _ from 'lodash';
+import { ListFacade } from '../src/facades/lists';
+import { Api } from '../src/api';
+import { ListValidator } from '../src/validators/lists';
+
+describe('Api', function() {
+  const tests = [
+    { fn: 'getList', validator: 'validateGetList', args: ['userId', 'apiKey', 'secretKey', 'itemType', 'listId'] },
+    { fn: 'getAllLists', validator: 'validateGetAllLists', args: ['userId', 'apiKey', 'secretKey', 'itemType'] },
+    { fn: 'getListItemCount', validator: 'validateGetListItemCount', args: ['userId', 'apiKey', 'secretKey', 'itemType', 'listId'] },
+    { fn: 'insertItemOnList', validator: 'validateItemOperation', args: ['userId', 'apiKey', 'secretKey', 'itemType', 'listId', 'itemId'] },
+    { fn: 'removeItemFromList', validator: 'validateItemOperation', args: ['userId', 'apiKey', 'secretKey', 'itemType', 'listId', 'itemId'] },
+    { fn: 'createList', validator: 'validateCreateList', args: ['userId', 'apiKey', 'secretKey', 'itemType', 'name'] },
+    { fn: 'updateList', validator: 'validateUpdateList', args: ['userId', 'apiKey', 'secretKey', 'itemType', 'listId', 'name'] },
+    { fn: 'deleteList', validator: 'validateDeleteList', args: ['userId', 'apiKey', 'secretKey', 'itemType', 'listId'] },
+  ];
+
+  let validatorSpy;
+  let facadeSpy;
+  let server;
+
+  before(function() {
+    server = sinon.createFakeServer();
+    server.respondWith([200, {}, 'OK']);
+    server.respondImmediately = true;
+  });
+
+  after(function() {
+    server.restore();
+  });
+
+  tests.forEach(function(test) {
```

Comment:
não acho bom colocar tests para serem feito nesse formato, seria bom realmente escrever um describe e it pra cada um

Reply:
Na verdade eu conversei com o @joao-fontenele e o @limafm e eles disseram que os testes poderia ser gerados dinamicamente uma vez que eles são exatamente iguais.